common.sh

```diff
@@ -536,10 +536,30 @@ EOF
 
 terminate_instances()
 {
-    # Terminates slave node
+    # Terminates compute node
     if [[ ! -z ${INSTANCE_IDS[@]} ]]; then
+        echo "Start terminating instances ${INSTANCE_IDS[@]}."
         AWS_DEFAULT_REGION=us-west-2 aws ec2 terminate-instances --instance-ids ${INSTANCE_IDS[@]}
-        AWS_DEFAULT_REGION=us-west-2 aws ec2 wait instance-terminated --instance-ids ${INSTANCE_IDS[@]}
+        local ret=0
+        local retry=5
+        # aws wait instance-terminated will poll every 15 seconds until a successful state has been reached.
+        # It will exit with a return code of 255 after 40 failed checks, i.e. 10 minutes. Retry this API call
+        # within $retry times in case some instances are not terminated within 10 minutes.
+        while [ "$retry" -ge 0 ]; do
+            AWS_DEFAULT_REGION=us-west-2 aws ec2 wait instance-terminated --instance-ids ${INSTANCE_IDS[@]}
```

Comment:
If `set -e` is in effect, would this lead to the script go to exit function?

Reply:
I see `set +e` in on_exit function https://github.com/awslabs/libfabric-ci-scripts/blob/master/common.sh#L502. But I could add a check for `-e` as I did in `get_instance_ip` as well.

Reply:
Thanks!

Reply:
Added this logic in the latest revision.